test/integration/magnetometer_system.cc

```diff
@@ -141,3 +141,146 @@ TEST_F(MagnetometerTest, GZ_UTILS_TEST_DISABLED_ON_WIN32(RotatedMagnetometer))
       field.Z(), TOL);
   mutex.unlock();
 }
+
+
+/////////////////////////////////////////////////
+// Test to check to ensure that the fix for using tesla units works correctly
+// Note(gilbert): It seems the above test (RotatedMagnetometer)
+// does not actually run because the spherical coordinates are not set.
+// In order to not break the above tests, we create a new world and run
+// two simulations, one with gauss and the other with tesla units.
+// See https://github.com/gazebosim/gz-sim/issues/2312
+
+class MagnetometerMessageBuffer {
+public:
+  void add(const msgs::Magnetometer &_msg)
+  {
+    std::lock_guard<std::mutex> lock(mutex);
+    magnetometerMsgs.push_back(_msg);
+  }
+
+  msgs::Magnetometer at(const size_t index) const
+  {
+    std::lock_guard<std::mutex> lock(mutex);
+    return magnetometerMsgs.at(index);
+  }
+
+  size_t size() const
+  {
+    std::lock_guard<std::mutex> lock(mutex);
+    return magnetometerMsgs.size();
+  }
+
+private:
+  mutable std::mutex mutex;
+  std::vector<msgs::Magnetometer> magnetometerMsgs;
+};
+
+MagnetometerMessageBuffer gaussMessages;
+MagnetometerMessageBuffer teslaMessages;
+
+/////////////////////////////////////////////////
+void gaussMagnetometerCb(const msgs::Magnetometer &_msg)
+{
+  gaussMessages.add(_msg);
+}
+
+/////////////////////////////////////////////////
+void teslaMagnetometerCb(const msgs::Magnetometer &_msg)
+{
+  teslaMessages.add(_msg);
+}
+
+TEST_F(MagnetometerTest,
+  GZ_UTILS_TEST_DISABLED_ON_WIN32(GaussToTeslaCorrection))
+{
+  // First, run the world without the fix enabled and record the magnetic field.
+  ServerConfig serverConfig;
+  const auto sdfFile = std::string(PROJECT_SOURCE_PATH) +
+    "/test/worlds/magnetometer_with_tesla.sdf";
```

Comment:
please use `gz::common::joinPaths`

Reply:
Great, thanks for the help. I updated the previous test as well.